client/js/components/snippets/snippet-form-dialog.jsx

```diff
@@ -31,16 +37,32 @@ export default class SnippetFormDialog extends React.Component {
       content: this.state.content,
       description: this.state.description,
       language: this.state.language
+    }).then(() => {
```

Comment:
Hmm I think it could be annoying to fill form over and over when someone encounters problem